share/wake/lib/system/job.wake

```diff
@@ -15,41 +15,111 @@
 
 package wake
 
+export tuple JobKey =
+    export dir: String
+    export stdin: String
+    export env: String
+    export cmd: String
+    export signature: Integer
+    export visible: String
+    export isatty: Integer
+
+# Create/reserve a job handle, parameters aren't necessarily finalized
+export def primJobCreate (label: String) ((JobKey dir stdin env cmd signature visible isatty): JobKey) (keep: Integer) (echo: String) (stdout: String) (stderr: String): Job =
```

Comment:
Likely personal preference (especially since these are internal-only functions), but it might be slightly cleaner to not deconstruct the JobKey on the top line:
```suggestion
export def primJobCreate (label: String) (jobKey: JobKey) (keep: Integer) (echo: String) (stdout: String) (stderr: String): Job =
    JobKey dir stdin env cmd signature visible isatty = jobKey

```